operator/pkg/validate/common.go

```diff
@@ -284,6 +284,11 @@ func UnmarshalIOP(iopYAML string) (*v1alpha1.IstioOperator, error) {
 		un.SetCreationTimestamp(meta_v1.Time{}) // UnmarshalIstioOperator chokes on these
 		iopYAML = util.ToYAML(un)
 	}
+	// TODO: find a better way to validate all the invalid values in IOP YAML
+	// before it is unmarshal into IstioOperator
+	if err := validateRevisionFromIOPYAML(iopYAML); err != nil {
```

Comment:
Why are we doing validation inside of `UnmarshalIOP`? This feels like the wrong layer to enter this. Don't we already have a common place for all of our validation?

Reply:
The validation runs after unmarshling to `IstioOperaror` which assumes revision passed is a string. But the problem is if invalid `revision` is passed eg:`18`, the unmarshling fails because `revision` must be string and not a number. This PR attempts to validate before unmarshling starts so we can detect invalid revision before unmarshling.

Reply:
I have already attempted to validate revision in validation step https://github.com/istio/istio/pull/28745 which only validates string revision. eg: "1.8.0".

Reply:
@howardjohn PTAL.

Reply:
@shamsher31 I understand why you put it here, but it doesn't feel correct to me. However, I also do not know a better place, as I don't work on this code base often.

Maybe @ostromart or @richardwxn can give a better review, I don't feel comfortable approving since I am not familiar and it isn't obviously correct to me. Thanks